### PR TITLE
feat(bicep): adding bicep support

### PR DESCRIPTION
**Pull request by JulioSCX**
Closes https://github.com/Checkmarx/kics/issues/6824
**Proposed Changes**
Adding support for Bicep files

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Review by cx-ruiaraujo — COMMENTED** [review]
Hi, please check my comments. Thanks

File: docs/future_improvements.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+**Note**: When using the "--disable-secrets" flag, be aware that this will disable the passwords and secrets query for all languages, not just Bicep files. As a result, you may miss some security checks in other files. Before using this flag, carefully consider the impact on your overall security coverage, especially if your project includes multiple languages or file types.
+
+We advise reviewing your project's specific security needs to determine if this flag is appropriate. More information about the passwords and secrets query is available in our [Password and Secrets documentation](https://github.com/Checkmarx/kics/blob/master/docs/secrets.md). 
+
+---
+
+## Contribution
+
+If you'd like to contribute or provide insightfull feedback regarding KICS' capabilities and limitations, please don't hesitate to contact [our team](https://github.com/Checkmarx/kics/issues/).
```
[the remark this answers is not in the document]
Comment (JulioSCX) [reply docs/future_improvements.md]: Thank you, fixed on the following commit https://github.com/Checkmarx/kics/pull/6980/commits/c39a1fdf26bbe2cf7db62aabff4a5755b840f493

File: docs/platforms.md [hunk 2]
```diff
@@ -9,14 +9,26 @@ KICS supports scanning Ansible files with `.yaml` extension.
...
+
+## Bicep 
+
+KICS supports scanning Bicep files with `.bicep` extension.
+
+For instructions on converting between ARM JSON templates and Bicep code, refer to the [official ARM documentation](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-cli#build). To understand the distinctions between the two, explore the [official syntax comparison](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/compare-template-syntax).
+
+Note that KICS recognizes this technology as Azure Resource Manager (for queries purpose).
+
+Explore our ongoing enhancements and planed features on our [Future Improvements](future_improvements.md) page.
```
[the remark this answers is not in the document]
Comment (JulioSCX) [reply docs/platforms.md]: Thank you, fixed on the following commit https://github.com/Checkmarx/kics/pull/6980/commits/c39a1fdf26bbe2cf7db62aabff4a5755b840f493

File: pkg/parser/bicep/parser.go [hunk 3]
```diff
@@ -0,0 +1,709 @@
...
+
+	lines, ok := resource[kicsLines].(map[string]interface{})
+	if !ok {
+		lines = map[string]interface{}{}
+	}
+
+	line := map[string]int{kicsLine: ctx.GetStart().GetLine()}
+	lines[kicsPrefix+"apiVersion"] = line
+
+	line = map[string]int{kicsLine: ctx.GetStart().GetLine()}
```
[the remark this answers is not in the document]
Comment (JulioSCX) [reply pkg/parser/bicep/parser.go]: Removed the line on this commit https://github.com/Checkmarx/kics/pull/6980/commits/f0203eaf38218f84323bebca8eefadc612a1eeda. Thank you.

File: go.mod [hunk 4]
```diff
@@ -22,7 +22,7 @@ require (
 	github.com/agnivade/levenshtein v1.1.1
 	github.com/alexmullins/zip v0.0.0-20180717182244-4affb64b04d0
 	github.com/antlr4-go/antlr/v4 v4.13.0
-	github.com/aws/aws-sdk-go-v2 v1.26.1
+	github.com/aws/aws-sdk-go-v2 v1.17.6
```
[the remark this answers is not in the document]
Comment (JulioSCX) [reply go.mod]: Upgraded back to 1.26.1, thank you.

File: pkg/analyzer/analyzer.go [hunk 5]
```diff
@@ -394,6 +396,12 @@ func (a *analyzerInfo) worker(results, unwanted chan<- string, locCount chan<- i
 				results <- terraform
 				locCount <- linesCount
 			}
+		// Bicep
+		case ".bicep":
```
[the remark this answers is not in the document]
Comment (JulioSCX) [reply pkg/analyzer/analyzer.go:400]: Added these tests on this commit https://github.com/Checkmarx/kics/pull/6980/commits/afecbfa76bf5db36bb38f0fb0ede2f59a6c9674f, thank you

File: pkg/parser/bicep/parser.go [hunk 6]
```diff
@@ -0,0 +1,709 @@
...
+
+// Adiciona as resources ao array "resources" do seu parent caso exista
+func addChildrenToParents(resources []*Resource) {
+	resourceMap := map[string]*Resource{}
+
+	for _, resource := range resources {
+		resourceMap[resource.Name] = resource
+	}
+
+	for _, resource := range resources {
```
[the remark this answers is not in the document]
Comment (JulioSCX) [reply pkg/parser/bicep/parser.go:118]: No, the map needs to have the information before going through it on the second loop, so the second loop is needed

File: pkg/analyzer/analyzer.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (ArturRibeiro-CX) [reply pkg/analyzer/analyzer.go:400]: We added a test to the analyzer in this [commit](027ad43061121d4f612275d8faf9f3276add2cb8), thanks!

**Review by cx-ruiaraujo — DISMISSED** [review]
LGTM! Thanks for the great work 💪

**Review by cx-ruiaraujo — DISMISSED** [review]

**Review by cx-andrep — APPROVED** [review]

**Pull request merged**